### DISCUSSION TITLE
Storing Passwords

### DISCUSSION THREAD

**Discussion by remchuk** [netbox-community/netbox#13242] [Q&A] [answered]
Is there a best practice for storing password to each device for in the NetBox?

There is a plugin for it but it is not good due to the fact that you can't store 1 password for many devices but rather for each device we need to configure a username and a device.

But is there a recommend way to do this?

**Comment by candlerb** [comment, marked as answer]
The Netbox secrets functionality (which was moved into a plugin) was not very good, for the reason you give.

I suggest you use a custom field to store a "password class" or "password name", which you in turn look up in a third-party system like Vault, Passbolt, Passwordstore etc.

**Reply by remchuk** [reply]
This is the best practice? Is there a reason for not having a good password storage in the netbox ? 

Like if i will suggest a feature, will it be considered ?

**Reply by kkthxbye-code** [reply]
> Like if i will suggest a feature, will it be considered ?

Very unlikely. The secrets functionality was intentionally removed.

https://github.com/netbox-community/netbox/issues/5278

As to the why, you can read the introduction here:

https://docs.netbox.dev/en/stable/introduction/

The `What NetBox Is Not` and `Keep It Simple` sections are relevant here. Security and encryption are complicated topics and there are a multitude of third party software that are better suited for it.

It is however perfectly feasible to create a plugin that supports whatever usecase you need, so I would suggest that as a solution for you.
